components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2238,11 +2238,34 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     public async findPrebuilds(ctx: TraceContext, params: FindPrebuildsParams): Promise<PrebuildWithStatus[]> {
         traceAPIParams(ctx, { params });
 
-        const user = this.checkAndBlockUser("getPrebuilds");
+        const user = this.checkAndBlockUser("findPrebuilds");
         await this.guardProjectOperation(user, params.projectId, "get");
         return this.projectsService.findPrebuilds(params);
     }
 
+    public async getPrebuild(ctx: TraceContext, prebuildId: string): Promise<PrebuildWithStatus | undefined> {
+        traceAPIParams(ctx, { prebuildId });
+        this.checkAndBlockUser("getPrebuild");
+        const pbws = await this.workspaceDb.trace({}).findPrebuiltWorkspaceById(prebuildId);
```

Comment:
Thanks for adding this one! :pray:
Instead of `trace({})` we should forward the `ctx` I think.

Reply:
Ah, and we still need some form of access restriction, as mentioned [here](https://github.com/gitpod-io/gitpod/pull/10696#discussion_r904965697).

Copying over the lines from the headless-log method should be fine. :+1:

Reply:
@geropl The existing access guards didn't work because they check for properties that the prebuildWithStatus didn't have, so I created a basic `GuardedPrebuild` resource. Tested the preview env and seems to work fine, but let me know what you think 🙏🏽

Reply:
Removed ✅